### PR TITLE
pass serviceId to all the components that need it and change data there depends on it

### PR DESCRIPTION
**Pull request by Hadeel-Salamin**
- [x]  Calendar component
 -  add the serviceId you got from the Link params.
 - change the endpoint from GET /events to /events/id.
 - get the service and events from the server response.
 - pass the **service id** to SideBar component as we need to show the details there.

- [x]  BookEvent component
 - get the serviceId from the Link params.
 - change the endpoint from POST /events to /events/id.
 - pass the **service id** to SideBar component.

- [x]  SideBar component
 - get the **service id** from props and use it to get services details by GET /events/id endpoint.
 - change the static value to show the service details you got from database.

- [x] getEvents BackEnd
 - add query to get service details with the same id.

- [x]  postEvent Backend
 - get serviceId from req params and add it to the event table "foreign key".
 - change the endpoint in the router file from POST /event to /event/serviceId.

- [x]  DetailsEvent component
 - get the serviceId from the event details.
 - pass the **service id** to SideBar component.

- [x] Login component
 - put the type of password input as **password**

- [x] general for all components 
 - make sure you can still go between pages with the right path.
 - give the service id to the path on details and book event components so once u  can cancel and approve you still can go back to the right calendar page.
 - change the paths in App.js file

### CODE REVIEW

File: client/src/components/BookEvent/index.js [hunk 1]
```diff
@@ -27,26 +26,32 @@ class BookEvent extends React.Component {
...
     axios
-      .post("/event", this.state)
+      .post(`/event/${serviceId}`, this.state)
       .then(data => {
-        history.push("/events");
+        history.push(`/events/${serviceId}`);
       })
-      .catch(error => console.log(error));
+      .catch(error => {
+        const { history } = this.props;
```
Comment (shahenazmonia) [inline client/src/components/BookEvent/index.js]: why did you identify ```history``` twice! you can use the first declaration for both

File: client/src/components/DetailsEvent/index.js [hunk 2]
```diff
@@ -35,25 +33,26 @@ class DetailsEvent extends Component {
   handleSubmitForm = event => {
     event.preventDefault();
     const { history } = this.props;
-
+    const { serviceId } = this.state.event_details;
     if (this.state.status === 0) {
       const id = this.props.history.location.event.id;
```
Comment (shahenazmonia) [inline client/src/components/DetailsEvent/index.js]: ```const {id} = this.props.history.location.event;``` clearer than before

File: client/src/components/BookEvent/index.js (hunk 1 above)
Comment (Hadeel-Salamin) [reply client/src/components/BookEvent/index.js]: removed

File: client/src/components/DetailsEvent/index.js (hunk 2 above)
Comment (Hadeel-Salamin) [reply client/src/components/DetailsEvent/index.js]: done

File: server/controllers/postEvent.js [hunk 3]
```diff
@@ -4,6 +4,7 @@ const {
 
 exports.postEvent = (req, res) => {
   const { userId } = req;
+  const { serviceId } = req.params;
```
Comment (shahenazmonia) [inline server/controllers/postEvent.js]: There is another way to identify variables by destructuring ```  const {
    params: { serviceId },
    userId,
    body: {
      title,
      start_date,
      end_date,
      org_name,
      price,
      capacity,
      note,
      coffee_note,
      coffee_price,
      coffee_time,
      lunch_note,
      lunch_price,
      lunch_time,
      equipment_price,
      equipment_note,
    },
  } = req; ```

**Pull request merged**